### PR TITLE
Delete the cert manager dependencies and generate the self-signed CA for webhook.

### LINKED ISSUE #1680: Double check and fixes the usage of cert-manager in vineyard-operator
We many noticed many failures and inconvenience about the cert-manager ...

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

As titled.

Related issue number
--------------------

Fixes #1680

### CODE REVIEW

File: k8s/config/crd/kustomization.yaml [hunk 1]
```diff
@@ -27,14 +27,14 @@ patchesStrategicMerge:
...
-- patches/cainjection_in_csidrivers.yaml
+#- patches/cainjection_in_localobjects.yaml
+#- patches/cainjection_in_globalobjects.yaml
+#- patches/cainjection_in_vineyardds.yaml
+#- patches/cainjection_in_operations.yaml
+#- patches/cainjection_in_sidecars.yaml
+#- patches/cainjection_in_backups.yaml
+#- patches/cainjection_in_recovers.yaml
+#- patches/cainjection_in_csidrivers.yaml
 # +kubebuilder:scaffold:crdkustomizecainjectionpatch
```
Comment (sighingnow) [inline k8s/config/crd/kustomization.yaml:38]: Remove those blocks if they are no longer needed...

File: k8s/config/default/kustomization.yaml [hunk 2]
```diff
@@ -37,34 +37,34 @@ patchesStrategicMerge:
...
+#    kind: Service
+#    version: v1
+#    name: webhook-service
+#  fieldref:
+#    fieldpath: metadata.namespace
+#- name: SERVICE_NAME
+#  objref:
+#    kind: Service
+#    version: v1
+#    name: webhook-service
```
Comment (sighingnow) [inline k8s/config/default/kustomization.yaml:70]: Ditto.

**Review by sighingnow — CHANGES_REQUESTED** [review]

File: k8s/config/crd/kustomization.yaml (hunk 1 above)
Comment (dashanji) [reply k8s/config/crd/kustomization.yaml:30]: Done.

File: k8s/config/default/kustomization.yaml (hunk 2 above)
Comment (dashanji) [reply k8s/config/default/kustomization.yaml]: Done.

**Review by sighingnow — APPROVED** [review]

**Pull request merged**